editor/editor_properties.cpp

```diff
@@ -1806,31 +1874,42 @@ EditorPropertyVector2::EditorPropertyVector2(bool p_force_wide) {
 	}
 	bc->set_h_size_flags(SIZE_EXPAND_FILL);
 
-	static const char *desc[2] = { "x", "y" };
-	for (int i = 0; i < 2; i++) {
+	spin_sliders.resize(component_count);
+	EditorSpinSlider **spin = spin_sliders.ptrw();
+
+	for (int i = 0; i < component_count; i++) {
 		spin[i] = memnew(EditorSpinSlider);
 		spin[i]->set_flat(true);
-		spin[i]->set_label(desc[i]);
+		spin[i]->set_label(String(COMPONENT_LABELS[i]));
 		bc->add_child(spin[i]);
 		add_focusable(spin[i]);
-		spin[i]->connect("value_changed", callable_mp(this, &EditorPropertyVector2::_value_changed).bind(desc[i]));
+		spin[i]->connect("value_changed", callable_mp(this, &EditorPropertyVectorN::_value_changed).bind(String(COMPONENT_LABELS[i])));
 		if (horizontal) {
 			spin[i]->set_h_size_flags(SIZE_EXPAND_FILL);
 		}
 	}
 
+	ratio.resize(component_count * (component_count - 1));
+	ratio.fill(1.0);
+
 	linked = memnew(TextureButton);
 	linked->set_toggle_mode(true);
 	linked->set_stretch_mode(TextureButton::STRETCH_KEEP_CENTERED);
-	linked->connect(SNAME("pressed"), callable_mp(this, &EditorPropertyVector2::_update_ratio));
+	linked->connect(SNAME("pressed"), callable_mp(this, &EditorPropertyVectorN::_update_ratio));
+	linked->connect(SNAME("toggled"), callable_mp(this, &EditorPropertyVectorN::_store_link));
 	hb->add_child(linked);
```

Comment:
I think that the button is missing a tooltip, preferably a dynamic one corresponding to its current state or reflecting what's going to happen when you click on it.